File: integer-to-roman/IntegerToRoman.java
class Solution {
    public String intToRoman(int num) {
        String[] numerals = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV","I"};
        int [] values ={1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};

        StringBuilder resultingSb = new StringBuilder("");
        int temp = num;
        
        for(int i = 0; i < numerals.length; i++){
            int repetition = temp / values[i]; // number of times numeral repeats successively    
            temp %= values[i];
            while(repetition > 0){
                resultingSb.append(numerals[i]);
                repetition--;
            }
        }

        return resultingSb.toString();
    }
}

File: integer-to-roman/IntegerToRoman.kt
class Solution {
    fun intToRoman(num: Int): String {
        var numerals = arrayOf("M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV","I")
        var values =arrayOf(1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1)
        var resultingSb = StringBuilder("")

        var temp = num
        for(i in numerals.indices){
            var repetition = temp / values[i]
            temp %= values[i]

            for(j in 0 until repetition){
                resultingSb.append(numerals[i])
            }
        }

        return resultingSb.toString()
    }
}